packages/fabric/src/index.ts

```diff
@@ -0,0 +1,169 @@
+import { Restroom } from "@restroom-mw/core";
+import { ObjectLiteral } from "@restroom-mw/core/src/types";
+import * as crypto from 'crypto';
+// How to include this?
+// import { combineDataKeys } from "@restroom-mw/sawroom/src/lib";
+import { Gateway, connect, Contract, Identity, Signer, signers,
+         Network }
+   from '@hyperledger/fabric-gateway';
+import { Client, credentials } from '@grpc/grpc-js';
+import axios from "axios";
+import { NextFunction, Request, Response } from "express";
+import {
+  ADDRESS,
+  CONNECT,
+  CHANNEL,
+  CONTRACT,
+  QUERY,
+  SUBMIT
+} from "./actions";
+import { TextDecoder } from 'util';
+
+let fabricAddress: string = null;
+let client: Client = null;
+let tlsCertificate;
+let input: ObjectLiteral = {};
+let gateway: Gateway = null;
+let network: Network = null;
+let contract: Contract = null;
+
+const utf8Decoder = new TextDecoder();
+
+
+// The steps in FabricInterop are all required (in this order...)
+enum FabricInterop {
+  Address = 0,
+  Connect,
+  Channel,
+  Contract
+}
+let current: FabricInterop = FabricInterop.Address;
+
+const validateStep = (requested: FabricInterop) => {
+  if(requested > current) {
+    throw new Error(
+`One step is missing
+1. Set endpoint address
+2. Connect organization
+3. Set channel
+4. Set contract (chaincode)`)
+  }
```

Comment:
Hmmm if they are all needed, and in order, why are not grouped in one sentence? 
I guess that will lead users to more errors.

Except obviously for the `onBefore` and `onSuccess` that should be on different sentences

Reply:
The order is not really mandatory, maybe in future...
I could do one statement but it would be really complex...

Reply:
So it's just for simplify the statements, in a more readable manner, correct?

Reply:
Yes